pkg/libb2/Makefile.libb2

```diff
@@ -2,6 +2,7 @@ MODULE = libb2
 
 # Compiling for native triggers this warning
 CFLAGS += -Wno-unused-function
+CFLAGS += -DSUFFIX=
```

Comment:
I saw this definition, but I thought that this is done by https://github.com/RIOT-OS/RIOT/blob/2101458900586d422ec20b63b16ee2757eab666f/pkg/libb2/include/config.h#L8-L9

Reply:
Furthermore, I have already the reimplementation of the ESP8266 port working (PR #11108). It uses gcc 5.2.0 compiled by myself while the current ESP8266 port uses gcc 4.8.5 which is provided by the vendor. With the reimplementation, I don't get this error.

All other platforms doesn't seem to have the problem.

So, it would be better to use this option only for the ESP8266. If it is OK for you, I would therefore try to extend the `cpu/esp866/Makefile.include` by an additional commit of my PR.

Reply:
Looks like that particular `config.h` doesn't get included, but some other `config.h` matches earlier.

If I rename this `config.h` to e.g. `libb2_config.h` and do so too in `blake2-impl.h` it works.

Reply:
Let me check which `config.h` is found earlier.

Reply:
Indeed, `$(NEWLIB)/xtensa-lx106-elf/include/config.h` is found instead of `$(RIOTBASE)/pkg/libb2/include/config.h`.

Reply:
The question is what the best way is to work around this problem. Extending ESP8266s make system by 
```diff
--- a/cpu/esp8266/Makefile.include
+++ b/cpu/esp8266/Makefile.include
@@ -105,6 +105,10 @@ else
     CFLAGS += -Os
 endif
 
+ifneq (, $(filter libb2, $(USEPKG)))
+    CFLAGS += -DSUFFIX=
+endif
+
```
works.

Reply:
Yea but that's ugly. And the package stays fragile - there are plenty of other `.config` files that might sneak in instead and cause unsuspected breakage.

Reply:
We should find another solution. Ignoring `pkg/libb2/config.h` completely, could also produce problems in future when this file is changed and contains definitions that are required.

What if we would rename `pkg/libb2/config.h/config.h` as you did and provide a patch file to include the renamed file. Providing a patch file to change something in original source code is the standard approach in RIOT packages, see section [Creating a patch with git](http://doc.riot-os.org/group__pkg.html). This would be the solution I would prefer.

Reply:
The second part of that config files is somewhat dubious too: It sets
```C
/* Test for a little-endian machine */
#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
#define NATIVE_LITTLE_ENDIAN
#endif
```

Now when checking where this is used, it's for constructs like
```C
#if defined(NATIVE_LITTLE_ENDIAN) && !defined(HAVE_ALIGNED_ACCESS_REQUIRED)
  return *( uint32_t * )( src );
#8424  
… (byte wise memory copying)
```

`HAVE_ALIGNED_ACCESS_REQUIRED` is never defined by us, so it will always use the potentially unaligned access on little endian machines.

Now the first best overview of what CPUs support unaligned loads was in [`gstconfig.h.in`](https://gitlab.freedesktop.org/gstreamer/gstreamer/blob/master/gst/gstconfig.h.in#L118) of the GStreamer source code:

```C
#if defined(__alpha__) || defined(__arc__) || defined(__arm__) || defined(__aarch64__) || defined(__bfin) || defined(__hppa__) || defined(__nios2__) || defined(__MICROBLAZE__) || defined(__mips__) || defined(__or1k__) || defined(__sh__) || defined(__SH4__) || defined(__sparc__) || defined(__sparc) || defined(__ia64__) || defined(_M_ALPHA) || defined(_M_ARM) || defined(_M_ARM64) || defined(_M_IA64) || defined(__xtensa__) || defined(__e2k__) || defined(__riscv)
#  define GST_HAVE_UNALIGNED_ACCESS 0
#elif defined(__i386__) || defined(__i386) || defined(__amd64__) || defined(__amd64) || defined(__x86_64__) || defined(__ppc__) || defined(__ppc64__) || defined(__powerpc__) || defined(__powerpc64__) || defined(__m68k__) || defined(_M_IX86) || defined(_M_AMD64) || defined(_M_X64) || defined(__s390__) || defined(__s390x__) || defined(__zarch__)
#  define GST_HAVE_UNALIGNED_ACCESS 1
#else
#  error "Could not detect architecture; don't know whether it supports unaligned access! Please file a bug."
#endif
```

And sure enough this will crash on the esp, but also on ARM (I tried `nucleo-l031k6`):

```C
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>

static inline uint32_t _unaligned_load(void* p) {
    return *(uint32_t*)p;
}

int main(void)
{
    puts("Hello World!");
    uint8_t* buffer = malloc(8);
    for (int i = 0; i < 8; ++i)
        buffer[i] = i;

    printf("unaligned load: %x\n", _unaligned_load(buffer + 1));

    return 0;
}
```